engine/runtime/src/main/java/org/enso/interpreter/runtime/error/PanicException.java

```diff
@@ -208,6 +215,50 @@ int getExceptionExitStatus() {
     return 1;
   }
 
+  @ExportMessage
+  boolean hasExceptionStackTrace() {
+    return true;
+  }
+
+  @ExportMessage
+  @CompilerDirectives.TruffleBoundary
+  Object getExceptionStackTrace(@Bind("$node") Node queryNode) throws UnsupportedMessageException {
+    var rawStack = TruffleStackTrace.getStackTrace(this);
+    var caughtIndex = findNodeIndexInStack(caughtBy, rawStack);
+    if (caughtIndex >= 0) {
+      var queryIndex = findNodeIndexInStack(queryNode, rawStack);
+      if (caughtIndex > queryIndex) {
+        // skip the part of stack between queryNode and caughtBy
+        var arr = new ArrayList<Object>();
+        for (var frame : rawStack.subList(0, queryIndex)) {
+          arr.add(frame.getGuestObject());
+        }
+        for (var frame : rawStack.subList(caughtIndex, rawStack.size())) {
+          arr.add(frame.getGuestObject());
+        }
+        return ArrayLikeHelpers.asVectorWithCheckAt(arr.toArray());
+      }
+    }
+    var fullArr = new ArrayList<Object>();
+    for (var frame : rawStack) {
+      fullArr.add(frame.getGuestObject());
+    }
+    return ArrayLikeHelpers.asVectorWithCheckAt(fullArr);
+  }
```

Comment:
It would be helpful if there was some explanation in the comment explaining why we need to 'cut' a part of the stack trace.

Reply:
There is no better explanation than description of #8153. It contains precise and enormous description of everything, including paragraph on why the stacktrace is wrong. Shall we copy such a text into the code?

I am not going to do it myself:
- there is going to be `git blame` link from the code to #12024 PR
- there is a link from the description of this PR to #8153 
- there is everything in #8153 (very long) description

While I believe all is sufficiently documented, I am not against contributions.

Reply:
I guess we could just add a link to this discussion then?

https://github.com/enso-org/enso/pull/12024#discussion_r1909850986

The blame can get lost after refactorings and is hard to find - having an explicit link may be better.

Reply:
OK, bf0a1b3